### ISSUE TITLE
Fix recursive function definition exploit

### ISSUE THREAD

**Issue by orcist** [tddbin/tddbin-frontend#23]
If a function is defined like `let foo = () => foo()` and then called `foo()`, the site lags in an infinite loop, some **maximum stack reached** error should be displayed perhaps instead?

**Comment by wolframkriesing** [comment]
yep, this would be nice to have handled, any suggestion on how to catch it?

**Comment by orcist** [comment]
A simple try-catch wrapper around the entire executed code should suffice, something like `try { foo(); bar(); baz(); } catch (e) { doSomethingWithException(e); }`

**Comment by wolframkriesing** [comment]
would have been nice, i didn't get that to catch it, I tried it in tddbin (just on the site)
if you have a better idea fire away, i am curious

**Comment by orcist** [comment]
Maybe replace Function.prototype.call() method with a try-catch wrapped version before executing function definition?
